tests/utils/paper_helpers.py

```diff
@@ -0,0 +1,18 @@
+import re
+
+def compare_formatted_citations(expected: str, actual: str) -> bool:
+    """
+    Compares two formatted citation strings; ignoring the citation_count value.
+
+    :param expected: The expected formatted citation string.
+    :param actual: The actual formatted citation string.
+    :return: True if the citations match except for the citation count, False otherwise.
+    """
+    # https://regex101.com/r/lCN8ET/1
+    citation_pattern = r"(This article has )\d+( citations?)"
+
+    # between group 1 and 2, replace with the character "n"
+    expected_cleaned = re.sub(citation_pattern, r"\1n\2", expected).strip()
+    actual_cleaned = re.sub(citation_pattern, r"\1n\2", actual).strip()
```

Comment:
I think you got the capturing groups wrong, you want `r"This article has (\d+) citations?"`

We don't care to extract the text, we want to extract the number

Then you can just directly compare the `int`, no need for the weird `r"\1n\2"`

Reply:
That's fair. 
This original approach using `\1n\2` was because it's focusing on comparing the structure of the citation text while ignoring the citation count itself; just ignoring whatever number is there.

But you make a valid point. I'll get this updated!

Reply:
Fwiw (I am not sure your experience levels with regex) going from `r"(This article has )\d+( citations?)"` to `r"This article has (\d+) citations?"` will still match the surrounding text too. A capturing group `()` is here to make it easy to capture and extract a value

And plz point out if we're on the same page already haha. Hope you have a good night

Reply:
I believe I've addressed this accordingly. Thx James!